packages/ragbits-cli/src/ragbits/cli/state.py

```diff
@@ -0,0 +1,64 @@
+import json
+from collections.abc import Sequence
+from dataclasses import dataclass
+from enum import Enum
+
+from pydantic import BaseModel
+from rich.console import Console
+from rich.table import Table
+
+
+class OutputType(Enum):
+    """Indicates a type of CLI output formatting"""
+
+    text = "text"
+    json = "json"
+
+
+@dataclass()
+class CliState:
+    """A dataclass describing CLI state"""
+
+    output_type: OutputType = OutputType.text
+
+
+cli_state = CliState()
+
+
+def print_output(data: Sequence[BaseModel] | BaseModel) -> None:
```

Comment:
in current state would rename it to `print_cli_output` or `cli_console_logger`

Reply:
Why? It's still under the `cli` namespace, but more importantly it simply prints to the standard output - just like Python's built-in `print` function. One could use it in non-cli contexts as wel.. On the other hand, "logger" has a very different connotation.

Reply:
because `print_output` looks very similar to visually to `print` and when somebody - potentially fresh with the codebase could get confused while reading it

Reply:
It works just like `print` but prints structured output. Isn't that enough? I can rename it to `print_cli` if you would prefer it. But it's a commonly used utility function, I don't want to be a long java-ish name from 3 components.